src/pages/multipleProductValidity.tsx

```diff
@@ -40,113 +39,148 @@ const MultipleProductValidity = ({
-                                            }`}
-                                        </td>
-                                        <td className="govuk-table__cell">
-                                            <span className="govuk-radios validity-select-wrapper">
-                                                <span className="govuk-radios__item">
+                                {multipleProducts.map((product, index) => {
+                                    return (
+                                        <tr className="govuk-table__row" key={product.productNameId}>
+                                            <td className="govuk-table__cell">{product.productName}</td>
+                                            <td className="govuk-table__cell">£{product.productPrice}</td>
+                                            <td className="govuk-table__cell">
+                                                {`${product.productDuration} ${product.productDurationUnits}${
+                                                    Number(product.productDuration) > 1 ? 's' : ''
+                                                }`}
+                                            </td>
+                                            <td className="govuk-table__cell">
+                                                <FormElementWrapper
+                                                    errors={errors}
+                                                    errorId={`validity-option-${index}`}
+                                                    errorClass="govuk-select--error"
+                                                    hideText
+                                                >
+                                                    <>
+                                                        {/* eslint-disable-next-line jsx-a11y/no-onchange */}
+                                                        <select
+                                                            className="govuk-select farestage-select"
+                                                            id={`validity-option-${index}`}
+                                                            name={`validity-option-${index}`}
+                                                            aria-labelledby={`stop-name-header stop-${index} naptan-code-header naptan-${index}`}
+                                                            onChange={handleSelection}
+                                                            defaultValue={product.productValidity}
+                                                        >
+                                                            <option selected value="" disabled>
+                                                                Select an expiry
+                                                            </option>
+                                                            <option value="24hr">24hr</option>
```

Comment:
Testing this page, found a few problems. If you select End of Service Day, and leave a box blank or complete it incorrectly re-renders the page, with no input boxes shown. Also, the same behaviour happens if you navigate back to the page - it does not automatically show the input boxes despite the End Of Service option being selected.